src/main/java/org/commcare/formplayer/services/MenuSessionFactory.java

```diff
@@ -106,6 +107,17 @@ public void rebuildSessionFromFrame(MenuSession menuSession, CaseSearchHelper ca
                         }
                     }
                 }
+                if (currentStep == null && processedStepsCount != steps.size()) {
```

Comment:
Could you push that in its own method? `rebuildSessionFromFrame` is quite long already

Reply:
[bf6e37e](https://github.com/dimagi/formplayer/pull/1560/commits/bf6e37e2230b0c66ff3b542a85d89a6219b4242b)